File: event-schemas/src/schemas/eventValues/txType.ts
/* eslint-disable @typescript-eslint/no-unused-vars */
import {
  JsonSchema, JsonSchemaType
} from '../../json-schema';

import { BoolFalseRef } from "../types";

export const TxType: JsonSchema = {
  title: 'TxType',
  type: JsonSchemaType.OBJECT,
  description: 'The TransactionType. All the values to set what kind of transaction is it',
  properties: {
    isAchievementReward: BoolFalseRef,
    isRedeem: BoolFalseRef,
    isReward: BoolFalseRef,
    isAdReward: BoolFalseRef,
    isInappPurchase: BoolFalseRef,
    isRenewal: BoolFalseRef,
    isPurchase: BoolFalseRef,
    isFXTrade: BoolFalseRef,
    isSale: BoolFalseRef,
    isTrade: BoolFalseRef,
    isDrop: BoolFalseRef,
    isLoot: BoolFalseRef,
    isGained: BoolFalseRef,
    isContract: BoolFalseRef,
  }
}

File: event-schemas/data/golang/TxType.go

package entities

// TxType represents a TxType model.
type TxType struct {
  IsAchievementReward bool
  IsRedeem bool
  IsReward bool
  IsAdReward bool
  IsInappPurchase bool
  IsRenewal bool
  IsPurchase bool
  IsFxTrade bool
  IsSale bool
  IsTrade bool
  IsDrop bool
  IsLoot bool
  IsGained bool
  IsContract bool
  AdditionalProperties map[string][]interface{}
}